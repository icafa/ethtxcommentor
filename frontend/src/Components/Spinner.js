import styled, { keyframes } from 'styled-components'
import { theme } from '@aragon/ui'

const donutSpin = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(359deg);
  }
`

export const SpinnerWrapper = styled.div`
  width: 100%;
  min-height: calc(100vh - 130px);
  margin: 0 auto;
  display: flex;
  justify-content: center;
  align-items: center;
`

const Spinner = styled.div`
  display: block;
  border: 4px solid rgba(0, 0, 0, 0.1);
  border-left-color: ${theme.accent};
  border-radius: 50%;
  width: 32px;
  height: 32px;
  animation: ${donutSpin} 1.2s linear infinite;
`

export default Spinner
