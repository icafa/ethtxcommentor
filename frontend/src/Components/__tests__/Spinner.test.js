import React from 'react'
import { render } from '@testing-library/react'
import Spinner from '../Spinner'

describe('Spinner Component', () => {
  it('renders correctly and matches snapshot', () => {
    const { asFragment } = render(<Spinner />)
    expect(asFragment()).toMatchSnapshot()
  })
})
