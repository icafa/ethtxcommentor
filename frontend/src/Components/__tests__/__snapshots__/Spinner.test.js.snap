// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-TOsTZ kbnHra"
  />
</DocumentFragment>
`;
